sherpa_contrib/diag_resp.py

```diff
@@ -637,13 +638,11 @@ def build_resp(emin, emax, offset: int, ethresh: Optional[Union[float,None]] = 1
         ### in sherpa/astro/instrument.py, or use a version test on
         ### whether this incrementation should be done
 
-        # from sherpa import __version__ as shpver
-        # major,minor,micro = shpver.split(".")
-        # ver = float(f"{major}.{minor}")
-        # ver_micro = int(micro)
-        #
+        major,minor,micro = shpver.split(".")
+        ver = float(f"{major}.{minor}")
+        #ver_micro = int(micro)
 
-        if offset != 1:
+        if ver < 4.17 and offset != 1:
```

Comment:
Technically this should not be necessary (as this should only ever be used with 4.17) but not worth changing.

Also, you can take advantage of tuple ordering rather than convert to a float - e.g

```
>>> ver = (4, 17)
>>> if ver < (4, 17):
...     print("Boo yah")
... 
>>> if ver <= (4, 17):
...     print("Boo yah")
... 
Boo yah
```

Again this doesn't need to be changed here.